src/rate.rs

```diff
@@ -1570,6 +1573,55 @@ impl RCState {
       .max(0)
       .min(cur_nframes_left - cur_scale_window_nframes)
   }
+
+  pub(crate) fn recv_frame_data_packet(
```

Comment:
Is there a place where this code originated from and can be factored out now?

Reply:
I couldn't find a good way to do that since it is intertwined with partial buffer management and the logic that returns the needed size.

Reply:
Maybe derf has some tips?

Reply:
I wouldn't spend time on it since the older API will go away sooner or later.

[here](https://github.com/xiph/rav1e/blob/master/src/rate.rs#L1607) if you want to have a look.